python/cudf/cudf/tests/test_datetime.py

```diff
@@ -642,21 +642,10 @@ def test_cudf_to_datetime(data, dayfirst):
     expected = pd.to_datetime(pd_data, dayfirst=dayfirst)
     actual = cudf.to_datetime(gd_data, dayfirst=dayfirst)
 
-    # TODO: Remove typecast to `ns` and following if/else
-    # workaround after following issue is fixed:
-    # https://github.com/pandas-dev/pandas/issues/52449
-
-    if actual is not None and expected is not None:
-        assert_eq(
-            actual.astype(pd_data.dtype)
-            if pd_data is not None
-            and hasattr(pd_data, "dtype")
-            and cudf.api.types.is_datetime_dtype(pd_data.dtype)
-            else actual.astype("datetime64[ns]"),
-            expected,
-        )
+    if isinstance(expected, pd.Series):
```

Comment:
Looks like this if/else branch uses the same `assert_eq` now

Reply:
Yes, but in our utility we branch out to `assert_series_equal` and `assert_index_equal` pandas utilities and if we mix up the function parameters either one of those will error. That's why this if/else branching.